### DISCUSSION TITLE
Table View Editor for Creating View instead of table

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#16919] [Contribute to Supabase]
# Feature request
Right now we only have Table Editor for concrete tables so we can directly view the data in supabase

## Is your feature request related to a problem? Please describe.
Problem is we do not have table view editor for `VIEWS`.
Typically we want to create complex table query like `relationship`, but doesn't scale well in the frontend code where you will add this complex queries to your code. So the solution is to create a custom view and add that SQL query so in the client side the complexity has been abstract.

## Describe the solution you'd like
Add a view editor page like table editor so we can view the output and navigate with paginate just like the table editor page.
We can also add a form page to create a view so we don't bloat the SQL page with different kinds of SQL Commands

**Comment by sandbox-apps** [comment]
And add a snippet for client side just like the snippet suggested for normal tables :D

**Comment by awalias** [comment]
> Add a view editor page like table editor so we can view the output and navigate with paginate just like the table editor page.

this one we already have

<img width="753" alt="image" src="https://user-images.githubusercontent.com/458736/125918021-3cb19e15-0810-4358-9d73-548b6403fbb3.png">

however as you mention, it is not yet possible to *edit* the view from the table editor, only read

**Comment by sandbox-apps** [comment]
Supabase is ultra fine even for beta <3

**Comment by nuurcodes** [comment]
Would be nice to be able to add RLS on views through the UI

**Comment by activenode** [comment]
@nuurcodes RLS cannot be addet to Views as to my knowledge - within supabase ATM.

I tried the proposed one here but it doesn't let me apply it https://www.cybertec-postgresql.com/en/view-permissions-and-row-level-security-in-postgresql/ 

That means any View will be `public` even if it is using protected data.

**Comment by dxrsz** [comment]
Short of a full view editor, it would be cool if there was just a pane when you open a view that displayed the output from:

```
SELECT VIEW_DEFINITION FROM INFORMATION_SCHEMA.VIEWS
   WHERE TABLE_NAME = '$current_view'
```

**Comment by MildTomato** [comment]
We've discussed this a few times and I think we ultimately figured that it would not make sense to have a GUI for editing/creating a view, but rather you write SQL for it. 

We could still provide some UI to help do this though, and make it easy, give prompts/hints on how to create and edit views.

**Reply by castortech** [reply]
At the very least a menu entry for VIEWS and the ability to list them and view their definitions.
